### PR TITLE
Report HCR progress whenever it succeeds or fails

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

It requires https://github.com/microsoft/java-debug/pull/335

### CODE REVIEW

**Comment by testforstephen** [comment]
When HCR succeeds, show a message at status bar for 5 seconds.
image

File: src/extension.ts [hunk 1]
```diff
@@ -178,14 +181,22 @@ async function applyHCR() {
...
-        }
-    });
+    if (!response || !response.changedClasses || !response.changedClasses.length) {
+        hcrStatusBar.clear();
+        vscode.window.showWarningMessage("Cannot find any changed classes for hot replace!");
+        return;
+    }
+
+    const changed = response.changedClasses.length;
+    hcrStatusBar.show("$(check)" + `${changed} changed classe${changed > 1 ? "s are" : " is"} reloaded!`, 5 * 1000);
```
Comment (jdneo) [inline src/extension.ts:199]: `changed class${changed > 1 ? "es are" : " is"}`

File: src/customWidget.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+
+export class NotificationBar {
+    private statusBar: vscode.StatusBarItem;
+    private lastUpdateTime: number;
+
+    constructor() {
+        this.statusBar = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, Number.POSITIVE_INFINITY);
```
Comment (jdneo) [inline src/customWidget.ts:11]: StatusBarItem is a disposable

**Comment by Eskibear** [comment]
> When HCR succeeds, show an inline message at status bar for 5 seconds

For me, I feel better to see a successful notification, consistent with what I see when it fails.

**Pull request merged**